doc/changelog/03-notations/11120-master+refactoring-application-printing.rst

```diff
@@ -0,0 +1,16 @@
+- **Fixed:** Inheritance of implicit arguments across notations made
+  uniform in parsing and printing. To the exception of notations of
+  the form :n:`Notation @string := @@qualid` and :n:`Notation @ident := @@qualid` which
+  inhibit implicit arguments, all notations binding a partially
+  applied constant, as e.g. in :n:`Notation @string := (@qualid @args)`,
+  or :n:`Notation @string := (@@qualid @args)`, or :n:`Notation @ident := (@qualid @args)`, or
+  :n:`Notation @ident := (@@qualid @args)`, inherit the remaining implicit arguments
```

Comment:
I didn't realize before but you can further simplify this list in just two:

```rst
as e.g. in :n:`Notation {| @string | @ident } := (@qualid @args)`
or :n:`Notation {| @string | @ident } := (@@qualid @args)`
```

or even in just one (maybe a bit extreme):

```rst
as e.g. in :n:`Notation {| @string | @ident } := ({? @ } @qualid @args)`
```

Reply:
I prefer not to make the reading to tricky.

Reply:
Apparently, sphinx fails on `@args` (`'token' reference target not found`). I replaced by `{+ @arg }` which seems to be the standard way to express what I meant (a non-empty list of arguments).